Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/BehaviourManagement/IssueBreachMitigation.hs

```diff
@@ -0,0 +1,158 @@
+
+import qualified Domain.Types.Common as DriverInfo
+import qualified Domain.Types.DriverBlockTransactions as DTDBT
+import qualified Domain.Types.DriverInformation as DI
+import qualified Domain.Types.Person as DP
+import Domain.Types.TransporterConfig
+import Kernel.Prelude
+import qualified Kernel.Storage.Hedis as Redis
+import Kernel.Types.Id
+import qualified Kernel.Types.SlidingWindowCounters as SWC
+import Kernel.Utils.Common
+import qualified Kernel.Utils.SlidingWindowCounters as SWC
+import Lib.Scheduler.Environment
+import Lib.Scheduler.JobStorageType.SchedulerType as JC
+import SharedLogic.Allocator
+import SharedLogic.BehaviourManagement.IssueBreach
+import qualified SharedLogic.External.LocationTrackingService.Flow as LTS
+import SharedLogic.External.LocationTrackingService.Types
+import Storage.Beam.SchedulerJob ()
+import qualified Storage.Queries.DriverInformation as QDriverInformation
+import Tools.Error
+import Tools.Metrics (CoreMetrics)
+
+mkIssueBreachCounterKey :: IssueBreachType -> Text -> Text
+mkIssueBreachCounterKey breachType driverId = "driver-offer:issue-breach:" <> show breachType <> "-counter-dId:" <> driverId
+
+mkCompletedBookingCounterKey :: IssueBreachType -> Text -> Text
+mkCompletedBookingCounterKey breachType driverId = "driver-offer:issue-breach:" <> show breachType <> ":completed-booking-counter-dId:" <> driverId
+
+incrementCompletedBookingCounterForIssueBreach ::
+  ( Redis.HedisFlow m r,
+    EsqDBFlow m r,
+    CacheFlow m r
+  ) =>
+  IssueBreachType ->
+  Id DP.Person ->
+  Integer ->
+  m ()
+incrementCompletedBookingCounterForIssueBreach breachType driverId windowSize = Redis.withCrossAppRedis $ SWC.incrementWindowCount (mkCompletedBookingCounterKey breachType driverId.getId) (SWC.SlidingWindowOptions windowSize SWC.Days)
+
+incrementIssueBreachCounter ::
+  ( Redis.HedisFlow m r,
+    EsqDBFlow m r,
+    CacheFlow m r
+  ) =>
+  IssueBreachType ->
+  Id DP.Person ->
+  Integer ->
+  m ()
+incrementIssueBreachCounter breachType driverId windowSize = Redis.withCrossAppRedis $ SWC.incrementWindowCount (mkIssueBreachCounterKey breachType driverId.getId) (SWC.SlidingWindowOptions windowSize SWC.Days)
+
+getIssueBreachCount ::
+  ( Redis.HedisFlow m r,
+    EsqDBFlow m r,
+    CacheFlow m r
+  ) =>
+  IssueBreachType ->
+  Integer ->
+  Id DP.Person ->
+  m Integer
+getIssueBreachCount breachType windowSize driverId = Redis.withCrossAppRedis $ SWC.getCurrentWindowCount (mkIssueBreachCounterKey breachType driverId.getId) (SWC.SlidingWindowOptions windowSize SWC.Days)
+
+getIssueBreachConfig :: IssueBreachType -> TransporterConfig -> Maybe IssueBreachConfig
+getIssueBreachConfig breachType transportConfig =
+  find (\issueBreachConfig -> issueBreachConfig.ibIssueBreachType == breachType) (fromMaybe [] (transportConfig.issueBreachConfig))
+
+getIssueBreachCooldownTime :: IssueBreachType -> DI.DriverInformation -> Maybe IssueBreachCooldownTime
```

Comment:
This file should only take TransporterConfig from Domain. 
DriverInformation and Person Domain type can be removed, it will help in reducing the Queries as those tables are not enabled in KV.